### ISSUE TITLE
Is it possible to use environment variables inside copy.bara.sky file?

### ISSUE THREAD

**Issue by scriptnull** [google/copybara#142]
Let us say that we have some information (like the commit header) in a environment variable of the system in which copybara runs (say `COMMIT_MESSAGE_ENV`)

Is it possible to refer to that environment variable in the metadata transformations.

```
core.workflow(
    name = "default",
    origin = git.github_pr_origin(
       ....
    ),

    destination = git.github_pr_destination(
       ....
    ),

   ....

    transformations = [
        metadata.add_header("COMMIT_MESSAGE_ENV should be present here"),
    ],
)
```

Currently, we are able to refer only the implicit labels like `GITHUB_PR_NUMBER`. But I wonder if there is a way to reference just the environment variables inside the copy.bara.sky file.

**Comment by hsudhof** [comment]
No, not at present. However, we have command line arguments for things like
the commit message. We are considering to add support for custom flags in
starlark, but this is still in motion.

Cheers, H

**Comment by yang** [comment]
@hsudhof Is there any other more general way to parameterize the workflows? Whether by environment variables, CLI arguments, or even input files.

Currently we dynamically generate copy.bara.sky files - is this the current recommended approach?

**Comment by mikelalcon** [comment]
No, we don't recommend to dynamically generate copybara files. Instead you should write a parameterized library and then create as many instances of workflows as needed. e.g.

copy.bara.sky:

def my_workflow(name, url, ...):
    core.workflow(
          name = name,
          url = url,
          ...
     )


my_workflow("foo1", "https://github.com/aaa/foo1")
my_workflow("foo2", "https://github.com/aaa/foo2")
my_workflow("foo3", "https://github.com/aaa/foo3")
my_workflow("foo4", "https://github.com/aaa/foo4")
my_workflow("foo5", "https://github.com/aaa/foo5")
